framework/src/base/NonlinearEigenSystem.C

```diff
@@ -71,6 +72,13 @@ NonlinearEigenSystem::solve()
   setInitialSolution();
   _time_integrator->solve();
   _time_integrator->postSolve();
+
+  // store eigenvalues
+  unsigned int n_converged_eigenvalues = getNumConvergedEigenvalues();
+  _eigen_values.clear();
+  _eigen_values.reserve(n_converged_eigenvalues);
```

Comment:
`clear()` followed by `reserve()` usually isn't the best pattern. I'd suggest that you just use `resize` instead just in case your system actually releases and reallocates memory.